contracts/migration/LiquidityMigrator.sol

```diff
@@ -2,92 +2,110 @@
+            deadline
         );
+        
+        // deposit the paired token into the new treasury
+        uint256 amountToken = token.balanceOf(address(this));
+        token.approve(address(newTreasury), amountToken);
+        newTreasury.deposit(amountToken, address(token), newTreasury.tokenValue(address(token), amountToken));
+
+        // burn the OHM
+        oldOHM.burn(oldOHM.balanceOf(address(this)));
+    }
 
-        // Migrate the V1 OHM to V2 OHM
-        oldOHM.approve(address(migrator), amountOHM);
-        migrator.migrate(amountOHM, IMigrator.TYPE.UNSTAKED, IMigrator.TYPE.UNSTAKED);
-        uint256 amountNewOHM = newOHM.balanceOf(address(this)).sub(initialNewOHMBalance); // # V1 out != # V2 in
+    function migrate(
+        uint256 amount,
+        IERC20 pair,
+        IUniswapV2Router router,
+        IERC20 token,
+        uint256 minToken,
+        uint256 minOhm,
+        uint256 deadline
+    ) external onlyGovernor {
+        // Fetch the treasury balance of the given liquidity pair
+        uint256 max = pair.balanceOf(address(oldTreasury));
+        uint256 lpAmount = amount < max ? amount : max;
+        oldTreasury.manage(address(pair), lpAmount);
 
-        // Add the V2 liquidity
-        IERC20(token).approve(address(routerTo), amountToken);
-        newOHM.approve(address(routerTo), amountNewOHM);
-        routerTo.addLiquidity(
-            token,
-            address(newOHM),
-            amountToken,
-            amountNewOHM,
-            amountToken,
-            amountNewOHM,
-            address(newTreasury),
-            _deadline
+        // Remove the liquidity
+        pair.approve(address(router), lpAmount);
+        router.removeLiquidity(
+            address(token),
+            address(oldOHM),
+            lpAmount,
+            minToken,
+            minOhm,
+            address(this),
+            deadline
         );
+        
+        // deposit the paired token into the new treasury
+        uint256 amountToken = token.balanceOf(address(this));
+
+        // burn the old OHM
+        uint256 amountOld = oldOHM.balanceOf(address(this));
+        oldOHM.burn(amountOld);
+
+        // mint new OHM after converting to new supply
+        uint256 amountNew = amountOld * newsohm.index() / oldsohm.index();
+        treasury.mint(address(this), amountNew);
 
-        // Send any leftover balance to the governor
-        newOHM.safeTransfer(authority.governor(), newOHM.balanceOf(address(this)));
-        oldOHM.safeTransfer(authority.governor(), oldOHM.balanceOf(address(this)));
-        IERC20(token).safeTransfer(authority.governor(), IERC20(token).balanceOf(address(this)));
+        newOHM.approve(address(router), amountNew);
+        token.approve(address(router), amountToken);
+
+        router.addLiquidity(
+            address(token), 
+            address(newOHM), 
+            amountToken, 
+            amountNew, 
+            amountToken * 99 / 100, 
+            amountNew * 99 / 100, 
```

Comment:
Why multiply by 99%?

Reply:
Also needs safemath

Reply:
min amounts could be passed in, but really all we need to ensure is it has not skewed significantly from the balance on the v1 side.